sys/shell/shell.c

```diff
@@ -66,160 +60,224 @@
     #define PROMPT_ON 0
 #endif /* SHELL_NO_PROMPT */
 
-static shell_command_handler_t find_handler(const shell_command_t *command_list, char *command)
-{
-    const shell_command_t *command_lists[] = {
-        command_list,
-        MORE_COMMANDS
-    };
+#ifdef MODULE_SHELL_COMMANDS
+    #define _builtin_cmds _shell_command_list
+#else
+    #define _builtin_cmds NULL
+#endif
 
-    /* iterating over command_lists */
-    for (unsigned int i = 0; i < ARRAY_SIZE(command_lists); i++) {
+#define SQUOTE '\''
+#define DQUOTE '"'
+#define ESCAPECHAR '\\'
+#define BLANK ' '
 
-        const shell_command_t *entry;
+int PARSE_ESCAPE_MASK = 0b100;
 
-        if ((entry = command_lists[i])) {
-            /* iterating over commands in command_lists entry */
-            while (entry->name != NULL) {
-                if (strcmp(entry->name, command) == 0) {
-                    return entry->handler;
-                }
-                else {
-                    entry++;
-                }
-            }
+enum parse_state {
+    PARSE_SPACE             = 0b000,
+
+    PARSE_UNQUOTED          = 0b001,
+    PARSE_SINGLEQUOTE       = 0b010,
+    PARSE_DOUBLEQUOTE       = 0b011,
+
+    PARSE_UNQUOTED_ESC      = 0b101,
+    PARSE_SINGLEQUOTE_ESC   = 0b110,
+    PARSE_DOUBLEQUOTE_ESC   = 0b111,
```

Comment:
`error: binary constants are a GCC extension [-Werror]`, change to hex?

Reply:
Oh ok, I didn't know that. I'll change it to hex then